File: Python/birthday-program/birthday-program.py
import os
import sys

thisYear = int(input("Please enter the current year "))
name = input("Please enter your name: ")
age = int(input("Please enter your year of birth: "))
choice = str(input("Do you already had birthday this year? yes/no "))

if choice == "yes":
    age = thisYear - age
    print("Hello " + name + "," + " you are " + str(age) + " years old")
else:
    age = thisYear - age - 1
    print("Hello " + name + "," + " you are still " + str(age) + " years old")

askAgainMessage = str(input("Would you like to ask another person? yes/no "))

if askAgainMessage == "yes":
    os.execv(sys.executable, ['python'] + sys.argv)
else:
    print("Good Bye!")

File: Go/birthday-program/birthday-program.go
package main

import (
	"fmt"
)

func main() {

	var name string
	var thisYear int
	var birthYear int
	var choice string

	fmt.Println("Please enter your name:")
	fmt.Scan(&name)
	fmt.Println("Please enter your year of birth")
	fmt.Scan(&birthYear)
	fmt.Println("Please specify this year")
	fmt.Scan(&thisYear)
	fmt.Println("Do you already had birthday this year? Y/N")
	fmt.Scan(&choice)

	if choice == "Y" {
		age := thisYear - birthYear
		fmt.Printf(fmt.Sprintf("Hello %v, you are %v years old", name, age))
	} else {
		age := thisYear - birthYear - 1
		fmt.Printf(fmt.Sprintf("Hello %v, you are still %v years old", name, age))
	}

	fmt.Println()
	var askAgainMessage string
	fmt.Println("Would you like to ask another person? Y/N")
	fmt.Scan(&askAgainMessage)

	if askAgainMessage == "Y" {
		main()
	} else {
		fmt.Printf("Good bye !")
	}
}